### ISSUE TITLE
Implement libuv-based Event Loop and Channel implementations

### ISSUE THREAD

**Issue by nayato** [Azure/DotNetty#53]

**Comment by AlgorithmsAreCool** [comment]
Would this work be similar to the high performance libuv work that [ASPNET has done for Kestrel](https://github.com/aspnet/KestrelHttpServer/tree/dev/src/Microsoft.AspNetCore.Server.Kestrel/Networking)?

**Comment by nayato** [comment]
@AlgorithmsAreCool, in some sense - yes. `KestrelThread` looks more or less aligned with `IEventExecutor` except that in Kestrel loop is only responsible for handling IO-related ops. Once IO is done, it dispatches to `IThreadPool`. Whereas in DotNetty one would also perform channel's work for processing IO completion results on the same thread. Here's what netty does:
https://github.com/netty/netty/blob/4.1/transport/src/main/java/io/netty/channel/nio/NioEventLoop.java#L337
Basically process IO, then take care of whatever's on the loop's queue.
I didn't look into it much but there's likely an interesting work of making sure byte buffers are working with libuv.

**Comment by nayato** [comment]
Consider broader solution including integration with Channels.
